src/main/java/com/google/cloud/tools/maven/CloudSdkOperationsFactory.java

```diff
@@ -23,19 +23,22 @@
 import com.google.common.base.Strings;
 
 public class CloudSdkOperationsFactory {
-  private String version;
 
-  CloudSdkOperationsFactory(String version) {
-    this.version = version;
+  // At the factory's creation time, the sdk version isn't configured yet, so we can't pass that in
+  // by itself. So we pass in a reference to the CloudSdkMojo and get the version when we need it.
+  private final CloudSdkMojo mojo;
+
+  CloudSdkOperationsFactory(CloudSdkMojo mojo) {
```

Comment:
we're only using version in here aren't we?

Reply:
The comment above explains it; mojo.getCloudSdkVersion() is still null at the time of the factory's instantiation, so I think we need a reference to the mojo, unless we want to instantiate the factory later on (which may make it trickier to test).

Reply:
Could the version be passed into `newManagedSdk`/`newChecker`?

Reply:
Maybe, would need to be passed into `newDownloader` as well since that uses `newChecker`. In that case though it would go back to what @loosebazooka and I were discussing earlier about needing tests to confirm the arguments are correctly used.

Reply:
Hmm, yea it just seems a bit off due to the cyclical dependency where CloudSdkMojo uses CloudSdkAppengineFactory which uses CloudSdkMojo.

Reply:
Pushed a change, let me know if it you think it needs tests.